### PR TITLE
PIM-10785: Fix case-insensitive patch product model

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by Balamung — APPROVED** [review]
Changelog ;)

**Review by hogren — APPROVED** [review]

File: tests/back/Pim/Enrichment/EndToEnd/Product/ProductModel/ExternalApi/PartialUpdateProductModelEndToEnd.php [hunk 1]
```diff
@@ -519,6 +519,30 @@ public function testUpdateRootProductModel()
         $this->assertEventCount(1, ProductModelUpdated::class);
     }
 
+    public function testUpdateCaseInsensitiveProductModel()
+    {
+        $client = $this->createAuthenticatedClient();
+
+        $data =
+            <<<JSON
+{
```
Comment (hogren) [inline tests/back/Pim/Enrichment/EndToEnd/Product/ProductModel/ExternalApi/PartialUpdateProductModelEndToEnd.php:528]:
```suggestion
        $data =
<<<JSON
{
```
Except for one test in this file, the standard is this. In the only one that differ and in other files, I found this :
```
        $data = <<<JSON
{
```

File: tests/back/Pim/Enrichment/EndToEnd/Product/ProductModel/ExternalApi/PartialUpdateProductModelEndToEnd.php [hunk 2]
```diff
@@ -519,6 +519,30 @@ public function testUpdateRootProductModel()
...
+        $client->request('PATCH', 'api/rest/v1/product-models/sweat', [], [], [], $data);
+
+        $response = $client->getResponse();
+
+        $this->assertSame(Response::HTTP_NO_CONTENT, $response->getStatusCode());
+        $this->assertArrayHasKey('location', $response->headers->all());
+        $this->assertSame(
+            'http://localhost/api/rest/v1/product-models/sweat',
+            $response->headers->get('location')
+        );
```
Comment (hogren) [inline tests/back/Pim/Enrichment/EndToEnd/Product/ProductModel/ExternalApi/PartialUpdateProductModelEndToEnd.php:543]: Just to be sure, can you double check that the first test is not redundant with the second ?

Comment (antoinearrive) [reply tests/back/Pim/Enrichment/EndToEnd/Product/ProductModel/ExternalApi/PartialUpdateProductModelEndToEnd.php]: I started again from the examples of tests below, I think this duplicate, I have just corrected, thank you

**Review by Beninho — APPROVED** [review]

**Pull request merged**